src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -621,6 +621,7 @@ class ObjectReferenceSegment(BaseSegment):
             Ref("BinaryOperatorGrammar"),
             Ref("ColonSegment"),
             Ref("DelimiterSegment"),
+            Ref("JoinLikeClauseGrammar"),
```

Comment:
@BorePlusPlus btw this is what I needed to add to stop the parser getting confused about whether `PIVOT` is an alias or a function.
Basically for an `ObjectReferenceSegment`, which is inherited by `TableReferenceSegment`, it will stop matching when the subsequent token is part of `JoinLikeClauseGrammar`, which contains the `FromPivotExpressionSegment` in Snowflake.